src/core-manager/index.js

```diff
@@ -135,12 +136,25 @@ export class CoreManager extends TypedEmitter {
         this.#handleExtensionMessage(data, peer)
       },
     })
+
+    this.#ready = Promise.all(
+      [...this.#coreIndex].map(({ core }) => core.ready())
+    ).catch(() => {})
```

Comment:
what happens if we try to use a core that failed to become ready later on? will we get a clear enough error during that time? wondering if it would make sense to do some handling here to catch earlier

Reply:
If a core failed to become ready then `coreManager.ready()` would reject.
If you try to use a core that failed to become ready then it would throw an error.
I hope the error we would get would be clear enough